lib/rules/no-mutation-props.js

```diff
@@ -0,0 +1,222 @@
+/**
+ * @fileoverview Prevent mutation of this.props
+ * @author Ian Schmitz, Joey Baker
+ */
+'use strict';
+
+const Components = require('../util/Components');
+const ARRAY_MUTATIONS = ['push', 'pop', 'shift', 'unshift'];
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent mutation of this.props',
+      category: 'Possible Errors',
+      recommended: false
+    }
+  },
+
+  create: Components.detect((context, components, utils) => {
+    /**
+     * Gets a unique ID for a node
+     * @param {Object} node The node to get an ID for
+     */
+    const getId = node => node && node.range.join(':');
+
+    const propsNodes = {};
+    /**
+     * Adds a node to the list of nodes that come from this.props
+     * @param {Object} node The node to add
+     */
+    const addToPropsNodes = node => {
+      const id = getId(node);
```

Comment:
`getId` can return a falsy value - that would get coerced to a string here, and probably provide undesired results.

Reply:
Sure? It would just try to find a falsey value key which would be `undefined`. I don't think there's any harm, but I agree it's safer and slightly faster to add a guard.

Reply:
Done.